.github/workflows/docker-backend.yml

```diff
@@ -87,27 +103,39 @@ jobs:
         run: |          
           # If it's a version tag for specific service, tag it with version
           if [[ ${{ github.ref_name }} =~ ^${{ inputs.name }}-v[0-9]+\.[0-9]+\.[0-9]+ ]]; then
+            if [[ $ {{ inputs.oss }} ]]; then
+              echo "REGISTRY=docker-oss.nexus.famedly.de" >> $GITHUB_ENV
+              echo "IMAGE_PATH=${REGISTRY}/${{ inputs.name }}" >> $GITHUB_ENV
+            fi
             VERSION=`echo "${{ github.ref_name }}" | sed -r "s/^${{ inputs.name }}-(v[0-9]+.[0-9]+.[0-9]+.*)$/\1/"`
-            docker tag build_image "${IMAGE_PATH}:${VERSION}"
+            docker tag build_image "${{ env.IMAGE_PATH }}:${VERSION}"
+          # If it's a nightly tag, use snapshot repo
+          elif  [[ ${{ github.ref_name  }} =~ ^(nightly|weekly)$  ]]; then
+            echo "REGISTRY=docker-nightly.nexus.famedly.de" >> $GITHUB_ENV
+            echo "IMAGE_PATH=${REGISTRY}/${{ inputs.name }}" >> $GITHUB_ENV
+            docker tag build_image "${{ env.IMAGE_PATH }}:${{ github.ref_name }}"
           # If it doesn't contain any version string, use it as-is
           elif ! [[ ${{ github.ref_name }} =~ v[0-9]+\.[0-9]+\.[0-9]+ ]]; then
-            docker tag build_image "${IMAGE_PATH}:${{ github.ref_name }}"
+            docker tag build_image "${{ env.IMAGE_PATH }}:${{ github.ref_name }}"
           fi
 
+
           # If it's a version tag without any additional release candidate / beta string etc., tag a latest release
           if [[ ${{ github.ref_name }} =~ ^${{ inputs.name }}-v[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
-            docker tag build_image "${IMAGE_PATH}:latest"
+            docker tag build_image "${{ env.IMAGE_PATH }}:latest"
           fi          
           # The rest will be version tags that are tagged for different service from monorepo
 
-      - name: Log into registry ${{ inputs.registry }}
+          docker tag build_image "${{ env.IMAGE_PATH }}:${{ github.sha }}"
+
+      - name: Log into registry ${{ env.registry }}
         uses: famedly/login-action@v2
         with:
-          registry: ${{ inputs.registry }}
+          registry: ${{ env.REGISTRY }}
           username: ${{ inputs.registry_user }}
           password: ${{ secrets.registry_password || secrets.GITHUB_TOKEN }}
 
       - name: Publish
         shell: bash
         run: |
-          docker image push --all-tags "${IMAGE_PATH}"
+          docker image push --all-tags "${{ env.IMAGE_PATH }}"
```

Comment:
Ah I see why you were doing GITHUB_ENV things ^^
Is it correct to only push to one repository? Can it happen in any way, that we push poth a snapshot to nightly and a release to releases? I think it might be.. So I think we need to push all tags for all repositories here right? :/

Reply:
Not in the same workflow run, I think? If it's a push to one of the named branches that triggers the workflow, the image will be pushed to the snapshot repo, and if it's triggered by a versioned tag, the releases/oss repo will be used